docker-compose.yml

```
@@ -1,185 +1,202 @@
-x-common-env: &common-env
-  FUZZ_RUNS: ${FUZZ_RUNS}
-  FUZZ_INPUT: ${FUZZ_INPUT}
-
 services:
   script:
     build:
       context: .
-      dockerfile: Dockerfile
-    environment:
-      <<: *common-env
-      CXXFLAGS: "-DBITCOIN_CORE -DRUST_BITCOIN"
-      FUZZ: script
+      tags:
+        - bitcoinfuzz:script
+      args:
+        CXXFLAGS: "-DBITCOIN_CORE -DRUST_BITCOIN"
+        FUZZ: script
+    env_file: .env
     volumes:
-      - ./docker/script:/app/data
+      - ./docker:/app/data
 
   deserialize_block:
     build:
       context: .
-      dockerfile: Dockerfile
-    environment:
-      <<: *common-env
-      CXXFLAGS: "-DBITCOIN_CORE -DRUST_BITCOIN -DBTCD"
-      FUZZ: deserialize_block
+      tags:
+        - bitcoinfuzz:deserialize_block
+      args:
+        CXXFLAGS: "-DBITCOIN_CORE -DRUST_BITCOIN -DBTCD"
+        FUZZ: deserialize_block
+    env_file: .env
     volumes:
-      - ./docker/deserialize_block:/app/data
+      - ./docker:/app/data
 
   script_eval:
     build:
       context: .
-      dockerfile: Dockerfile
-    environment:
-      <<: *common-env
-      CXXFLAGS: "-DBITCOIN_CORE -DBTCD"
-      FUZZ: script_eval
+      tags:
+        - bitcoinfuzz:script_eval
+      args:
+        CXXFLAGS: "-DBITCOIN_CORE -DBTCD"
+        FUZZ: script_eval
+    env_file: .env
     volumes:
-      - ./docker/script_eval:/app/data
+      - ./docker:/app/data
 
   descriptor_parse:
     build:
       context: .
-      dockerfile: Dockerfile
+      tags:
+        - bitcoinfuzz:descriptor_parse
+      args:
+        CXXFLAGS: "-DBITCOIN_CORE -DNBITCOIN -DRUST_MINISCRIPT -DEMBIT"
+        FUZZ: descriptor_parse
+    env_file: .env
     environment:
-      <<: *common-env
-      CXXFLAGS: "-DBITCOIN_CORE -DNBITCOIN -DRUST_MINISCRIPT -DEMBIT"
-      FUZZ: descriptor_parse
-      ASAN_OPTIONS: detect_leaks=0
+      LIBFUZZ_DETECT_LEAKS: 0
     volumes:
-      - ./docker/descriptor_parse:/app/data
+      - ./docker:/app/data
 
   kernel_block:
     build:
       context: .
-      dockerfile: Dockerfile
+      tags:
+        - bitcoinfuzz:kernel_block
+      args:
+        CXXFLAGS: "-DPYBITCOINKERNEL -DRUSTBITCOINKERNEL"
+        FUZZ: kernel_block
+    env_file: .env
     environment:
-      <<: *common-env
-      CXXFLAGS: "-DPYBITCOINKERNEL -DRUSTBITCOINKERNEL"
-      FUZZ: kernel_block
       ASAN_OPTIONS: detect_leaks=0
     volumes:
-      - ./docker/kernel_block:/app/data
+      - ./docker:/app/data
 
   miniscript_parse:
     build:
       context: .
-      dockerfile: Dockerfile
+      tags:
+        - bitcoinfuzz:miniscript_parse
+      args:
+        CXXFLAGS: "-DBITCOIN_CORE -DNBITCOIN -DRUST_MINISCRIPT -DEMBIT"
+        FUZZ: miniscript_parse
+    env_file: .env
     environment:
-      <<: *common-env
-      CXXFLAGS: "-DBITCOIN_CORE -DNBITCOIN -DRUST_MINISCRIPT -DEMBIT"
-      FUZZ: miniscript_parse
-      ASAN_OPTIONS: detect_leaks=0
+      LIBFUZZ_DETECT_LEAKS: 0
```

For all targets with a module using the JVM, we need to set ASAN_OPTIONS: detect_leaks=0

```suggestion
environment:
      LIBFUZZ_DETECT_LEAKS: 0
      ASAN_OPTIONS: detect_leaks=0
```

---

Still missing to add the flags.

---

Done in a73e0e65f3e653e04ee300b360d7dee48478e4ef